MediaBrowser.Common/Net/NetworkExtensions.cs

```diff
@@ -9,240 +13,351 @@ namespace MediaBrowser.Common.Net
+                        }
+                    }
                 }
             }
 
-            return false;
+            return cidrnet;
         }
 
         /// <summary>
-        /// Returns true if the collection contains an item with the ip address,
-        /// or the ip address falls within any of the collection's network ranges.
+        /// Converts an IPAddress into a string.
+        /// Ipv6 addresses are returned in [ ], with their scope removed.
         /// </summary>
-        /// <param name="source">The <see cref="Collection{IPObject}"/>.</param>
-        /// <param name="item">The item to look for.</param>
-        /// <returns>True if the collection contains the item.</returns>
-        public static bool ContainsAddress(this Collection<IPObject> source, IPObject item)
+        /// <param name="address">Address to convert.</param>
+        /// <returns>URI safe conversion of the address.</returns>
+        public static string FormatIpString(IPAddress? address)
         {
-            if (source.Count == 0)
+            if (address is null)
             {
-                return false;
+                return string.Empty;
             }
 
-            ArgumentNullException.ThrowIfNull(item);
-
-            foreach (var i in source)
+            var str = address.ToString();
+            if (address.AddressFamily == AddressFamily.InterNetworkV6)
             {
-                if (i.Contains(item))
+                int i = str.IndexOf('%', StringComparison.Ordinal);
+                if (i != -1)
                 {
-                    return true;
+                    str = str.Substring(0, i);
                 }
+
+                return $"[{str}]";
             }
 
-            return false;
+            return str;
         }
 
         /// <summary>
-        /// Compares two Collection{IPObject} objects. The order is ignored.
+        /// Try parsing an array of strings into <see cref="IPNetwork"/> objects, respecting exclusions.
+        /// Elements without a subnet mask will be represented as <see cref="IPNetwork"/> with a single IP.
         /// </summary>
-        /// <param name="source">The <see cref="Collection{IPObject}"/>.</param>
-        /// <param name="dest">Item to compare to.</param>
-        /// <returns>True if both are equal.</returns>
-        public static bool Compare(this Collection<IPObject> source, Collection<IPObject> dest)
+        /// <param name="values">Input string array to be parsed.</param>
+        /// <param name="result">Collection of <see cref="IPNetwork"/>.</param>
+        /// <param name="negated">Boolean signaling if negated or not negated values should be parsed.</param>
+        /// <returns><c>True</c> if parsing was successful.</returns>
+        public static bool TryParseToSubnets(string[] values, out List<IPNetwork> result, bool negated = false)
         {
-            if (dest is null || source.Count != dest.Count)
+            result = new List<IPNetwork>();
```

Comment:
Is a list necessary?

Reply:
We do dynamically add items later on in the code so we can't use an array directly.

Reply:
You can use IReadOnlyList/Collection and assign the list to a local variable and then assign it to `result` before you return.